src/semantic_memory/utils.py

```
@@ -0,0 +1,79 @@
+        json.dump(data, f, ensure_ascii=False, indent=indent)
+
+
+def simple_token_count(text: str) -> int:
+    """Very naive token count: split on whitespace.
+    In production replace with a proper tokenizer (e.g., tiktoken).
+    """
+    return len(text.split())
+
+
+def cosine_similarity(vec1, vec2) -> float:
+    """Compute cosine similarity between two equal‑length iterables of numbers.
```

Ruff (0.15.14)

[warning] Docstring contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)